### PR TITLE
Add GitHub action

### PR DESCRIPTION
**Pull request by yl-1993**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+            torchvision: 0.6.0
+            cuda_arch: "7.0"
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install CUDA
```
Comment (hellock) [inline .github/workflows/build.yml:54]: We do not need to install CUDA.

Comment (yl-1993) [reply .github/workflows/build.yml]: CUDA and mmcv_full has been removed.

File: requirements/tests.txt [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+codecov
+flake8
+interrogate
+isort
+# Note: used for kwarray.group_items, this may be ported to mmcv in the future.
+kwarray
+pytest
+pytest-cov
+pytest-runner
+ubelt
```
Comment (hellock) [inline requirements/tests.txt:11]: `kwarray`, `pytest-cov`, `pytest-runner` and `ubelt` are unnecessary.

Comment (yl-1993) [reply requirements/tests.txt]: unnecessary packages have been removed

File: setup.py [hunk 3]
```diff
@@ -105,7 +178,13 @@ def get_requirements(filename='requirements.txt'):
...
-        tests_require=['pytest', 'xdoctest'],
-        install_requires=get_requirements(),
+        setup_requires=parse_requirements('requirements/build.txt'),
+        tests_require=parse_requirements('requirements/tests.txt'),
+        install_requires=parse_requirements('requirements/runtime.txt'),
+        extras_require={
+            'all': parse_requirements('requirements.txt'),
+            'tests': parse_requirements('requirements/tests.txt'),
+            'build': parse_requirements('requirements/build.txt'),
+            'optional': parse_requirements('requirements/optional.txt'),
```
Comment (hellock) [inline setup.py:188]: no optional.txt

Comment (yl-1993) [reply setup.py]: 'optional' has been removed in setup.py

**Pull request merged**